src/components/Modal.astro

```diff
@@ -0,0 +1,94 @@
+---
+interface Props {
+  modalContent: {
+    title?: string;
+    description?: string;
+  };
+}
+
+const {
+  modalContent: { title, description },
+} = Astro.props;
+---
+
+<div
+  id="modal"
+  class="fixed hidden top-1/2 left-1/2 max-w-2xl w-full -translate-x-1/2 -translate-y-1/2"
+  data-show="false"
+>
+  <div
+    class="mx-8 p-6 relative flex flex-col bg-sky-200 border-sky-400 border-2 md:p-10"
+  >
```

Comment:
```suggestion
<div class="fixed inset-0 hidden" data-visible="false" id="modal">
  <div class="absolute inset-0 backdrop-blur-sm z-10"></div>
  <div class="absolute inset-0 z-20 flex justify-center items-center">
    <div class="max-w-2xl w-full">
      <div
        class="modal-container mx-8 p-6 relative flex flex-col bg-sky-200 border-sky-400 border-2 md:p-10"
      >
```

Reply:
Notice the `.modal-container`, you'll need it for onOutsideClick

Reply:
changed